src/Easy.zig

```diff
@@ -111,6 +111,89 @@ pub const Response = struct {
                 .name = name,
             };
     }
+
+    pub const HeaderIterator = struct {
+        handle: *c.CURL,
+        name: ?[:0]const u8,
+        request: ?usize,
+        c_header: ?*polyfill_struct_curl_header() = null,
+
+        pub fn next(self: *HeaderIterator) !?Header {
+            if (comptime !has_parse_header_support()) {
+                return error.NoCurlHeaderSupport;
+            }
+
+            const request: c_int = if (self.request) |v| @intCast(v) else -1;
+
+            if (self.c_header == null) {
+                // Slightly more efficient than `curl_easy_nextheader`.
+                if (self.name) |filter_name| {
+                    try checkCode(c.curl_easy_header(
+                        self.handle,
+                        filter_name.ptr,
+                        0,
+                        c.CURLH_HEADER,
+                        request,
+                        &self.c_header,
+                    ));
+                    const c_header = self.c_header orelse unreachable;
+                    return Header{
+                        .c_header = c_header,
+                        .name = std.mem.sliceTo(c_header.*.name, 0),
+                    };
+                }
+            }
+
+            while (true) {
+                const c_header = c.curl_easy_nextheader(
```

Comment:
I think `curl_easy_header` has the ability to get all headers with same name with `index` param.

```
CURLHcode curl_easy_header(CURL *easy,
                           const char *name,
                           size_t index,
                           unsigned int origin,
                           int request,
                           struct curl_header **hout);
```

Reply:
`curl_easy_header()` searches for the nth name from beginning of all headers everytime.

`curl_easy_nextheader()` searches from the `prev` header.

Reply:
Fair point